### PR TITLE
BAEL-6846 - Converting from Float to BigDecimal in Java

### PR DESCRIPTION
**Pull request by mcasari**
Added a class with unit test cases for BAEL-6846 - Converting from Float to BigDecimal in Java. 

I added the class to the sub-module /master/core-java-modules/core-java-numbers-conversions, since it seemed the most natural choice to me.

In some some of the previous commits I simply removed the module I added before for my test article and a temporary module I was using for the present article.

### CODE REVIEW

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-numbers-conversions/src/test/java/com/baeldung/floattobigdecimal/FloatToBigDecimalUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        String floatValue = Float.toString(1.1f);
+        BigDecimal bdFromString = new BigDecimal(floatValue);
+        assertEquals("1.1", bdFromString.toString());
+    }
+
+    @Test
+    public void whenCreatedByValueOfAndIsFloat_thenDoesNotMatch() {
+        float floatToConvert = 1.1f;
+        BigDecimal bdByValueOf = BigDecimal.valueOf(floatToConvert);
+        assertNotEquals("1.1", bdByValueOf.toString());
```
[the remark this answers is not in the document]
Comment (mcasari) [reply core-java-modules/core-java-numbers-conversions/src/test/java/com/baeldung/floattobigdecimal/FloatToBigDecimalUnitTest.java]: Ok, done

File: core-java-modules/core-java-numbers-conversions/src/test/java/com/baeldung/floattobigdecimal/FloatToBigDecimalUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        float floatToConvert = 0.5f;
+        BigDecimal bdFromFloat = new BigDecimal(floatToConvert);
+        assertEquals("0.5", bdFromFloat.toString());
+    }
+
+    @Test
+    public void whenCreatedFromCertainFloatValues_thenDoesNotMatch() {
+        float floatToConvert = 1.1f;
+        BigDecimal bdFromFloat = new BigDecimal(floatToConvert);
+        assertNotEquals("1.1", bdFromFloat.toString());
```
[the remark this answers is not in the document]
Comment (mcasari) [reply core-java-modules/core-java-numbers-conversions/src/test/java/com/baeldung/floattobigdecimal/FloatToBigDecimalUnitTest.java]: Ok, done

**Review by theangrydev — APPROVED** [review]

**Comment by mcasari** [comment]
I changed the test cases to align them to the article updates, based on the discussions about the float internal representation.

File: core-java-modules/core-java-numbers-conversions/src/test/java/com/baeldung/floattobigdecimal/FloatToBigDecimalUnitTest.java [hunk 3]
```diff
@@ -16,45 +16,44 @@ public void whenFloatComparedWithDifferentValues_thenCouldMatch() {
...
     @Test
-    public void whenFloatCastToDouble_thenGotADifferentNumber() {
-        float floatToConvert = 1.1f;
-        double doubleCast = floatToConvert;
-        assertEquals("1.100000023841858", Double.toString(doubleCast));
+    public void whenCreatedByValueOf_thenFloatValueGetsTruncated() {
+        assertEquals("1.100000023841858", BigDecimal.valueOf(1.1f)
+            .toString());
+        assertEquals("1.100000023841858", BigDecimal.valueOf(1.10000002384185791015625f)
+            .toString());
```
Comment (theangrydev) [inline core-java-modules/core-java-numbers-conversions/src/test/java/com/baeldung/floattobigdecimal/FloatToBigDecimalUnitTest.java:50]: Can we format each of these on a single line, or do you prefer it the way it is?

Comment (mcasari) [reply core-java-modules/core-java-numbers-conversions/src/test/java/com/baeldung/floattobigdecimal/FloatToBigDecimalUnitTest.java]: Ok, I fixed the format.

Comment (mcasari) [reply core-java-modules/core-java-numbers-conversions/src/test/java/com/baeldung/floattobigdecimal/FloatToBigDecimalUnitTest.java]: I changed the test cases a little to align with the article updates

Comment (mcasari) [reply core-java-modules/core-java-numbers-conversions/src/test/java/com/baeldung/floattobigdecimal/FloatToBigDecimalUnitTest.java]: I changed the string constructor test case with the clearer "1.1" input value

**Review by theangrydev — APPROVED** [review]

**Pull request merged**